hazelcast/src/main/java/com/hazelcast/sql/impl/expression/string/StringExpressionUtils.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.Row;
+
+import java.util.Locale;
+
+/**
+ * Utility methods for string functions.
+ */
+public final class StringExpressionUtils {
+    private StringExpressionUtils() {
+        // No-op.
+    }
+
+    public static String concat(String first, String second) {
+        return (first != null ? first : "") + (second != null ? second : "");
+    }
+
+    public static Integer position(String seek, String source, int position) {
+        if (seek == null) {
+            return null;
+        }
+
+        if (source == null) {
+            return null;
+        }
+
+        if (position == 0) {
+            return source.indexOf(seek) + 1;
+        } else {
+            int position0 = position - 1;
+
+            if (position0 < 0 || position0 > source.length()) {
+                return 0;
+            }
+
+            return source.indexOf(seek, position0) + 1;
+        }
+    }
+
+    public static String substring(String source, Integer startPos, Integer length) {
+        // TODO: Validate the implementation against ANSI.
+        if (source == null || startPos == null || length == null) {
+            return null;
+        }
+
+        int sourceLength = source.length();
+
+        if (startPos < 0) {
+            startPos += sourceLength + 1;
+        }
+
+        int endPos = startPos + length;
+
+        if (endPos < startPos) {
+            throw QueryException.error("End position is less than start position.");
+        }
+
+        if (startPos > sourceLength || endPos < 1) {
+            return "";
+        }
+
+        int startPos0 = Math.max(startPos, 1);
+        int endPos0 = Math.min(endPos, sourceLength + 1);
+
+        return source.substring(startPos0 - 1, endPos0 - 1);
+    }
+
+    public static Integer charLength(String value) {
+        return value != null ? value.length() : null;
+    }
+
+    public static Integer ascii(String value) {
+        return value != null ? value.isEmpty() ? 0 : value.codePointAt(0) : null;
```

Comment:
that's not ascii, that's unicode code point

Reply:
What is the problem with it? Will it return incorrect values for some ASCII characters? 

The behavior of this function is not well-defined for non-ASCII characters. The only common thing across databases is that it should return the value that could be converted back to the same character through `CHR/NCHAR/etc` function. This will be possible with `Character.toChars()` when it is time to implement `CHR` function.

Perhaps a simple cast should also work fine.